### PR TITLE
[google_maps_flutter] Widget based polyline support for google maps.

### PR DESCRIPTION
**Pull request by awazgyawali**
PR to support polyline on both Android and IOS.

### **UPDATE: Mar 9, 2019**
This PR initially was a controller based support. As the flutter team has added widget based marker support, so I also added the same widget based support for the polyline.

If you want to use controller-based polyline support use the following
```
 google_maps_flutter: 
    git:
      url: git://github.com/awazgyawali/plugins/
      path: packages/google_maps_flutter
      ref: cd06409
```

and if you want to use the widget based solution use the following.
```
  google_maps_flutter:
      git:
        url: git://github.com/awazgyawali/plugins/
        path: packages/google_maps_flutter
        ref: google_maps_polyline
```

### CODE REVIEW

**Comment by awazgyawali** [comment]
@irevans Can you sign the CLA agreement anytime soon? I think you were about to confirm it from your employer.

**Comment by ZatacaSystems** [comment]
@irevans Any news on this?

**Comment by ZatacaSystems** [comment]
> @irevans Can you sign the CLA agreement anytime soon? I think you were about to confirm it from your employer.

@irevans Any news about this?

**Comment by irevans** [comment]
@awazgyawali I'm still waiting on my employer to sign the cla, it's shouldn't take much longer, i'm chasing it ip regularly.

When you added the iOS support I was wondering whether you took a look at the flutter code. I'm interested in whether you think using null to represent the default pattern in the PolylineOptions class is a good idea. Perhaps using an empty list would be better, such that null is used to consistently mean "do not change this configuration option".

**Comment by awazgyawali** [comment]
Never noticed that coz IOS sdk for google maps doesn't seem to have support for patterns out of the box nor start cap and end cap, so I didn't look at the pattern code.

However I found this “Dotted Polylines with Google Maps SDK for iOS” by Dylan Maryk https://link.medium.com/bjo68I3AwT just now.

I might have to look at this and your work to implement pattern for android to check if they match or not.

**Comment by irevans** [comment]
@awazgyawali I'm going to running the iOS version of the sample application. Where is the google maps API KEY specified on the iOS side?

**Comment by awazgyawali** [comment]
Check AppDelegate.m file.

**Comment by irevans** [comment]
@awazgyawali FYI, my employer has signed the cla. I have added a "I signed it" comment on my original fork but it didn't trigger a rebuild. I'm not sure why.

**Comment by awazgyawali** [comment]
@irevans Not sure but I think you have to mention "I signed it" on this thread too.

**Comment by irevans** [comment]
I signed it!

**Comment by awazgyawali** [comment]
Yes, I am aware of the issue. Although setting width 10 for both platforms, Google maps SDK seems to paint different thickness on different platform. For example, if I set 10 width on ios it paints a thinker line than 10 width on android. I think this is an issue on the unit measurement system of android and IOS platform. 

One thing we can do is add/subtract the thickness on one platform to match the width on both, but I think this would not be a good idea to do.

**Comment by irevans** [comment]
@awazgyawali Hmm, thats a bit of a problem in my mind. A developer will want their polylines to be rendered the same on both platforms. Not sure how best to address that, maybe some guidance from the flutter team would be best.  

Does iOS support different joint types? it doesn't appear to change no matter which type I choose?

Have you also seen that maps are centred differently on iOS and Android?

**Comment by awazgyawali** [comment]
Maybe @zoechi can mention someone here who can guide us.

**Comment by irevans** [comment]
@awazgyawali wrt the joint type, I can see that on iOS it doesn't look like the google maps sdk supports joint type.

What about the geodesic property, it doesn't look like you're doing anything with that but it looks like it is supported on iOS?

**Comment by awazgyawali** [comment]
@irevans  geodesic property should work on IOS too as I have implemented that on IOS side.

**Comment by irevans** [comment]
@awazgyawali the googlebot is now happy with the CLA on #941. If you trigger a rebuild of this pull request it should be OK now too.

**Comment by awazgyawali** [comment]
@irevans I think you have to mention "I signed it" here too.

**Comment by irevans** [comment]
I signed it!

**Comment by irevans** [comment]
@awazgyawali I noticed that since you forked my repo I have made a few additional commits. It's worth you merging these changes into your repo.

**Comment by irevans** [comment]
@awazgyawali I am using your branch in my application. Can you merge the latest changes from irevans/plugins as there is a minor fix on that branch that you will want to include here?

**Comment by awazgyawali** [comment]
Sure, when I get time.

**Comment by awazgyawali** [comment]
@amirh What do you think about this PR, many people are interested in polyline support. Do you think it should wait for the parameter based change in the plugin? Either way, let me know, I can help on making this feature on the parameter way too.

**Comment by nandymandy1** [comment]
I want to get a coordinates of location on a tapped point and place the marker can I get demo codes for that??? With flutter Google maps plugins

**Comment by irevans** [comment]
@amirh I would like to echo what @awazgyawali said. There a lot of people who would like to see this PR merged. What do you think?

**Comment by rlecheta** [comment]
Hi, I used the Polyline and it worked like a charm. But my project needs Polygon. Can I use it or it is not done yet?

**Comment by awazgyawali** [comment]
@rlecheta There is no support for polygon but I have another PR with circle support if that helps. 
https://github.com/flutter/plugins/issues/1136

**Comment by amirh** [comment]
Thanks again for putting this together, this is a really great contribution!
I'm sorry it takes so long, I've commented on https://github.com/flutter/flutter/issues/17832#issuecomment-465327497 with a more detailed status update.

**Comment by DebkanchanSamadder** [comment]
Hi guys,
I wanted to use your polyline support as it'll be the only last necessary bit in our app and without it it's nothing, so tried to install it by adding this in my `pubspec.yaml`

```
  google_maps_flutter: 
    git:
      url: git://github.com/awazgyawali/plugins/
      path: packages/google_maps_flutter
      ref: google_maps_polyline
```

but it throws this error when I try to run `flutter packages get`

```
Git error. Command: git rev-list --max-count=1 google_maps_polyline
fatal: ambiguous argument 'google_maps_polyline': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

**Comment by awazgyawali** [comment]
I have changed the way plugin works for polyline to the widget way as marker works on current branch. If you want to use the previous technique, please use the following snippet.
```
 google_maps_flutter: 
    git:
      url: git://github.com/awazgyawali/plugins/
      path: packages/google_maps_flutter
      ref: cd06409
```

For the newer change, you can use
```
 google_maps_flutter: 
    git:
      url: git://github.com/awazgyawali/plugins/
      path: packages/google_maps_flutter
      ref: google_maps_polyline
```

Note: Its a work in progress and works only on Android. IOS on its way.

**Comment by DebkanchanSamadder** [comment]
I tried again with 

```
dependencies:
  flutter:
    sdk: flutter
  google_maps_flutter: 
    git:
      url: git://github.com/awazgyawali/plugins/
      path: packages/google_maps_flutter
      ref: google_maps_polyline
```

it still throws the same
```
Git error. Command: git rev-list --max-count=1 google_maps_polyline

fatal: ambiguous argument 'google_maps_polyline': unknown revision or path not in the working tree.

Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

am I doing it right?

**Comment by DebkanchanSamadder** [comment]
> I tried again with
> 
> ```
> dependencies:
>   flutter:
>     sdk: flutter
>   google_maps_flutter: 
>     git:
>       url: git://github.com/awazgyawali/plugins/
>       path: packages/google_maps_flutter
>       ref: google_maps_polyline
> ```
> it still throws the same
> 
> ```
> Git error. Command: git rev-list --max-count=1 google_maps_polyline
> 
> fatal: ambiguous argument 'google_maps_polyline': unknown revision or path not in the working tree.
> 
> Use '--' to separate paths from revisions, like this:
> 'git <command> [<revision>...] -- [<file>...]'
> ```
> am I doing it right?

Turns out you need to add `.git` at the end of the path like this

```
google_maps_flutter: 
    git:
      url: git://github.com/awazgyawali/plugins.git
      path: packages/google_maps_flutter
      ref: google_maps_polyline
```

**Comment by awazgyawali** [comment]
I have completed the transfer of the polyline support to widget-based similar to the marker on from the core team. Details on how to use the PR on controller basis and widget basis is mentioned on the description of this PR.

@amirh I am aware that the plugin team is busy on more important topics. Just letting you know this update with this mention.

**Comment by irevans** [comment]
@awazgyawal Great job getting the conversion done. I'll look to test it early in the coming week. By the way I readded my employer to the Authors list, who I contribute on behalf of.

**Comment by cwichoski** [comment]
Any estimation for this feature to be merged in official plugin?

**Comment by awazgyawali** [comment]
Check this out https://github.com/flutter/flutter/issues/17832#issuecomment-465327497.

**Comment by awazgyawali** [comment]
Great work @irevans. I might have missed that.

**Comment by ened** [comment]
Looks like a fantastic PR. Are the marker changes required to make polyline support work?

@amirh @iskakaushik would you have time to review this PR? What needs to happen to increase chances of inclusion?

**Comment by iskakaushik** [comment]
Hi @awazgyawali, Thanks so much for this PR. I'm working on adding support for integration tests for google maps plugin at https://github.com/flutter/plugins/pull/1347. We are tracking this PR as a high priority item and we want to make sure that we add new features with integration tests in place.

**Comment by awazgyawali** [comment]
Hello @iskakaushik, thanks for the update about the test. I will check the #1347 and maybe even try to write the test for polyline too.

File: packages/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java
Comment (iskakaushik) [inline packages/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java]: Add `result.success(null)`. I forgot to do it for markers, if you rebase to the tip of master you should see that change :)

File: packages/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/PolylineBuilder.java
Comment (iskakaushik) [inline packages/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/PolylineBuilder.java]: we need to update `this.consumeTapEvents`

File: packages/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/PolylineController.java
Comment (iskakaushik) [inline packages/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/PolylineController.java]: need to update `this.consumeTapEvents`

File: packages/google_maps_flutter/example/ios/Runner.xcodeproj/project.pbxproj
Comment (iskakaushik) [inline packages/google_maps_flutter/example/ios/Runner.xcodeproj/project.pbxproj]: don't checkin xcode file changes.

File: packages/google_maps_flutter/ios/Classes/GoogleMapPolylineController.h
Comment (iskakaushik) [inline packages/google_maps_flutter/ios/Classes/GoogleMapPolylineController.h]: remove blank lines.

File: packages/google_maps_flutter/example/ios/Runner.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist
Comment (iskakaushik) [inline packages/google_maps_flutter/example/ios/Runner.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: remove this file

File: packages/google_maps_flutter/example/ios/Runner.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings
Comment (iskakaushik) [inline packages/google_maps_flutter/example/ios/Runner.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings]: remove this file

File: packages/google_maps_flutter/example/ios/Runner/Base.lproj/Main.storyboard
Comment (iskakaushik) [inline packages/google_maps_flutter/example/ios/Runner/Base.lproj/Main.storyboard]: remove this file

File: packages/google_maps_flutter/example/lib/place_marker.dart
Comment (iskakaushik) [inline packages/google_maps_flutter/example/lib/place_marker.dart]: remove marker drag stuff from this PR.

File: packages/google_maps_flutter/lib/src/cap.dart
Comment (iskakaushik) [inline packages/google_maps_flutter/lib/src/cap.dart]: Please refer https://www.dartlang.org/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph . Start with one line description, blank line and details. For this and sub sequent comments.

File: packages/google_maps_flutter/lib/src/google_map.dart
Comment (iskakaushik) [inline packages/google_maps_flutter/lib/src/google_map.dart]: `///` instead of `//`

File: packages/google_maps_flutter/lib/src/cap.dart
Comment (iskakaushik) [inline packages/google_maps_flutter/lib/src/cap.dart]: annotate this class with `@immutable`. That way you don't need to mention the immutable nature as a comment and dart analyzer will lint useful stuff, for example [this](https://dart-lang.github.io/linter/lints/prefer_const_constructors_in_immutables.html)

File: packages/google_maps_flutter/lib/src/pattern_item.dart
Comment (iskakaushik) [inline packages/google_maps_flutter/lib/src/pattern_item.dart]: `@immutable` and fix comment.

File: packages/google_maps_flutter/lib/src/pattern_item.dart
Comment (iskakaushik) [inline packages/google_maps_flutter/lib/src/pattern_item.dart]: this is not a really useful comment. Let's change this to:

```
/// A dash used in the stroke pattern for a [Polyline].
///
/// [length] has to be non-negative.
```

please fix this for `gap` as well.

File: packages/google_maps_flutter/lib/src/polyline.dart
Comment (iskakaushik) [inline packages/google_maps_flutter/lib/src/polyline.dart]: I am guessing that this comment was before `PolylineOptions` were made part of `Polyline`. Please fix this to reflect the current nature of the widget.

File: packages/google_maps_flutter/lib/src/polyline.dart
Comment (iskakaushik) [inline packages/google_maps_flutter/lib/src/polyline.dart]: Please fix the comments in the file per the suggestion regarding effective dart documentation reference above.

**Review by iskakaushik — CHANGES_REQUESTED** [review]
@awazgyawali, great job overall, thanks for the PR. We have now added support for on-device tests for google maps plugin. [Please take a look at this file for an example test case](https://github.com/flutter/plugins/pull/1409/files#diff-2e2be5700c933ace6962002d886f09b0). I will be happy to help you through any issue you run into while adding tests. Please add some relevant tests to your PR :-)

I've also left some comments. I've looked at dart and android side of the changes. I'm guessing that the comments left on android side are also relevant to the ios impl. Please address those as well.

I will keep an eye out for any updates you make to this PR. Thanks again!

File: packages/google_maps_flutter/lib/src/cap.dart
Comment (awazgyawali) [reply packages/google_maps_flutter/lib/src/cap.dart]: @irevans Can you please work on this, as you are the author to code it would be easy for you to do. :-)

Comment (irevans) [reply packages/google_maps_flutter/lib/src/cap.dart]: @awazgyawali Sure, just to be clear, shall I resolve all the issues raised by @iskakaushik or are you planning on looking at some?

Comment (awazgyawali) [reply packages/google_maps_flutter/lib/src/cap.dart]: I am done with some changes. WIll be pushing changes in some hours. Can you look after that?

**Review by iskakaushik — CHANGES_REQUESTED** [review]

File: packages/google_maps_flutter/example/ios/Runner.xcodeproj/project.pbxproj
Comment (irevans) [reply packages/google_maps_flutter/example/ios/Runner.xcodeproj/project.pbxproj]: @awazgyawali Don't think I have permission to mark these as resolved. I suggest you take a look at my commit and resolve the issue if you agree with the changes.

File: packages/google_maps_flutter/lib/src/cap.dart
Comment (irevans) [reply packages/google_maps_flutter/lib/src/cap.dart]: See commit 1455ca0

**Comment by irevans** [comment]
@awazgyawali Following the changes you made to migrate to a widget based approach it looks like a number of parameters in the Polyline type now no longer have default values. I noticed that's causing one of your tests to fail (because color is null). I would suggest the following change. WDYT?

`const Polyline({
    @required this.polylineId,
    this.consumeTapEvents = false,
    this.color **= Colors.black**,
    this.endCap **= Cap.buttCap**,
    this.geodesic = false,
    this.jointType **= JointType.mitered**,
    this.points = const <LatLng>[],
    this.pattern **= const <PatternItem>[]**,
    this.startCap **= Cap.buttCap**,
    this.visible = true,
    this.width = 1,
    this.zIndex = 0,
    this.onTap,
  });`

**Comment by awazgyawali** [comment]
@irevans I was trying to add default variables before but wasn't able to decide which values to use. The list above is fine I guess. I suggest we add that now and let googles suggest a change if needed.

**Comment by irevans** [comment]
> @irevans I was trying to add default variables before but wasn't able to decide which values to use. The list above is fine I guess. I suggest we add that now and let googles suggest a change if needed.

Ok I'll go ahead and make those changes.

**Comment by irevans** [comment]
@awazgyawali  I noticed you changed the default polyline width to 1. Any reason for that? I originally chose 10 as it reflected the default in the underlying native maps sdk. Note that the comments still say that it defaults to 10. If there is a good reason to default to 1 then we should update the comments.

**Comment by awazgyawali** [comment]
Let's make it 10, I guess that's an optimal value. Also, can you check the issues with the formatter?

File: packages/google_maps_flutter/example/ios/Runner.xcodeproj/project.pbxproj
Comment (irevans) [reply packages/google_maps_flutter/example/ios/Runner.xcodeproj/project.pbxproj]: See commit a1982e4. I believe it should resolve this issue.

File: packages/google_maps_flutter/ios/Classes/GoogleMapController.h
Comment (iskakaushik) [inline packages/google_maps_flutter/ios/Classes/GoogleMapController.h]: Is there a reason this method is added?

Comment (irevans) [reply packages/google_maps_flutter/ios/Classes/GoogleMapController.h]: @awazgyawali looks like you added this one. Could you take a look at this?

Comment (awazgyawali) [reply packages/google_maps_flutter/ios/Classes/GoogleMapController.h]: I don't recall it either, wasn't it already on the master branch. I will check if this function is used and remove it.

File: packages/google_maps_flutter/ios/Classes/GoogleMapController.m
Comment (cyanglaz) [inline packages/google_maps_flutter/ios/Classes/GoogleMapController.m]: nit: this is my personal opinion, feel free to ignore. Declare as `id` is not helpful to understand what `args[@"polylinesToAdd"]` should be expected to be. I had to look the code below to see that you are checking the type to know that you are expecting an NSArray. You would still need the type check below, but at least by reading this line, I know what to expect from the `args[@"polylinesToAdd"]`.
```suggestion
    NSArray* polylinesToAdd = args[@"polylinesToAdd"];
```
Same as below, from line 136 to line 149

File: packages/google_maps_flutter/ios/Classes/GoogleMapController.m
Comment (cyanglaz) [inline packages/google_maps_flutter/ios/Classes/GoogleMapController.m]: Should we assert and throw if this check fails? Same questions to below from line 136 to line 149.

File: packages/google_maps_flutter/ios/Classes/GoogleMapPolylineController.m
Comment (cyanglaz) [inline packages/google_maps_flutter/ios/Classes/GoogleMapPolylineController.m]: Can `zIndex`, `points`, `strokeColor`, `strokeWidth` have different types? If their types are known, declare them as specific type instead of `id`

File: packages/google_maps_flutter/ios/Classes/JsonConversions.m
Comment (cyanglaz) [inline packages/google_maps_flutter/ios/Classes/JsonConversions.m]: Just out of curiosity, how does casting an NSArray to an NSNumber work?

**Review by cyanglaz — COMMENTED** [review]
Only reviewed the ObjectiveC changes. Left some comments.

File: packages/google_maps_flutter/ios/Classes/GoogleMapController.m
Comment (awazgyawali) [reply packages/google_maps_flutter/ios/Classes/GoogleMapController.m]: I tried to consistent with the existing logic to add markers. For example, in line  88 `markersToAdd` is also stored as id. I would love to change polyline one to the NSArray but not sure what to do with the marker. Can you suggest how should I move on?

File: packages/google_maps_flutter/ios/Classes/JsonConversions.m
Comment (awazgyawali) [reply packages/google_maps_flutter/ios/Classes/JsonConversions.m]: Not sure either, I just sent an integer from dart code which was then casted to NSArray on Obj-C. Upon searching and testing, found out that NSArray can be casted to NSNumber,

File: packages/google_maps_flutter/ios/Classes/GoogleMapPolylineController.m
Comment (awazgyawali) [reply packages/google_maps_flutter/ios/Classes/GoogleMapPolylineController.m]: Done.

File: packages/google_maps_flutter/test/polyline_updates_test.dart
Comment (irevans) [reply packages/google_maps_flutter/test/polyline_updates_test.dart]: @iskakaushik you're happy for an issue to be raised before the pull request is merged?

Comment (iskakaushik) [reply packages/google_maps_flutter/test/polyline_updates_test.dart]: Yup, let's file it preemptively.

File: packages/google_maps_flutter/ios/Classes/GoogleMapController.h
Comment (awazgyawali) [reply packages/google_maps_flutter/ios/Classes/GoogleMapController.h]: Removed.

File: packages/google_maps_flutter/test/polyline_updates_test.dart
Comment (awazgyawali) [reply packages/google_maps_flutter/test/polyline_updates_test.dart]: Filed one here https://github.com/flutter/flutter/issues/30764

Comment (awazgyawali) [reply packages/google_maps_flutter/test/polyline_updates_test.dart]: Change the issue id to the newer one.

**Comment by cujslive** [comment]
There are problems with Android Pie. App is crashing when map opens.
Following is the error details. @awazgyawali Please note. Let me know if you need anything more.

```
java.lang.NoClassDefFoundError: Failed resolution of: Lorg/apache/http/ProtocolVersion;
	at fb.b(:com.google.android.gms.dynamite_mapsdynamite@16089081@16.0.89 (100400-239467275):3)
	at fa.a(:com.google.android.gms.dynamite_mapsdynamite@16089081@16.0.89 (100400-239467275):3)
	at fc.a(:com.google.android.gms.dynamite_mapsdynamite@16089081@16.0.89 (100400-239467275):15)
	at com.google.maps.api.android.lib6.drd.al.a(:com.google.android.gms.dynamite_mapsdynamite@16089081@16.0.89 (100400-239467275):6)
	at ee.a(:com.google.android.gms.dynamite_mapsdynamite@16089081@16.0.89 (100400-239467275):23)
	at ee.run(:com.google.android.gms.dynamite_mapsdynamite@16089081@16.0.89 (100400-239467275):8)
Caused by: java.lang.ClassNotFoundException: Didn't find class "org.apache.http.ProtocolVersion" on path: DexPathList[[zip file "/data/user_de/0/com.google.android.gms/app_chimera/m/0000000e/MapsDynamite.apk"],nativeLibraryDirectories=[/data/user_de/0/com.google.android.gms/app_chimera/m/0000000e/MapsDynamite.apk!/lib/arm64-v8a, /system/lib64]]
	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:171)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
	at ac.loadClass(:com.google.android.gms.dynamite_dynamiteloader@16089081@16.0.89 (100400-239467275):4)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
	... 6 more
```

**Comment by iskakaushik** [comment]
@cujslive This issue has been fixed in newer versions of https://github.com/flutter/plugins/blob/master/packages/google_maps_flutter/CHANGELOG.md#0303

**Comment by iskakaushik** [comment]
@awazgyawali looks like some of the functionality is still missing on the iOS side, caps, joints etc. Do you plan on adding it?

**Comment by awazgyawali** [comment]
@iskakaushik I am aware of the missing features. I was planning to add joints feature on IOS but haven't found a way to actually change the cap. Also for the joints, the SDKs seems to provide different kind of methods on the IOS side. Not sure how we should use them. 

On android it has `PatternItem` but on IOS it has Spans. I think we can use spans too to make the gap behavior work by using transparent color. Details are [here](https://developers.google.com/maps/documentation/ios-sdk/shapes#change_the_color_of_individual_line_segments).

**Comment by iskakaushik** [comment]
@awazgyawali ah, i see. I made the example more obvious to account for that. I also added some dart doc on the fields that are currently supported on Android only. We can add some version of translation from `patterns` to `spans` on iOS. I do not think that should block this PR.

**Review by iskakaushik — APPROVED** [review]

**Comment by iskakaushik** [comment]
For CLA, i've manually verified that the only committers are:

```
Author: Aawaz Gyawali <awazgyawali@gmail.com>
Author: EVANS, Ian 3 <Ian.Evans3@admiralgroup.co.uk>
Author: Ian Evans <ian.evans3@admiralgroup.co.uk>
Author: Kaushik Iska <kaushikiska@google.com>
```

My suspicion is that the bot is not case insensitive. Overriding the CLA check.

**Comment by irevans** [comment]
@iskakaushik would I be right in thinking that only the user that opened the PR gets contribution credit for this work?

**Comment by iskakaushik** [comment]
@irevans , i'm not sure what you mean by contribution credit. I will check with my team and let you know. If there is some explicit credit, i'll make sure that you get it as well.

**Comment by irevans** [comment]
@iskakaushik I was referring to githubs recognition that a user is a contributor to a project. I found this in GitHub's documentation [here](https://help.github.com/en/articles/why-are-my-contributions-not-showing-up-on-my-profile):

"When a pull request is merged and commits are squashed, only the user that merged the pull request and the user that opened the pull request receive contribution credit. No other contributors to the pull request will receive contribution credit."

In which case I'm guessing there is nothing you can do about that. Thanks in any case.

**Comment by stevenspiel** [comment]
@awazgyawali Have you given thought to Polygon support? Do you think they could coexist? Structurally, what makes the most sense to you?

* An abstract `Polyshape` class that `Polyline` and `Polygon` both implement
* A `Polyshape` object with an `enclosed` bool option (polygons would also need a fillColor param)
* Handle them separately by duplicating `Polyline` code and adjusting where needed to allow for polygon support

I'd love to know your thoughts on it.

**Comment by thiagotalma** [comment]
Is it possible to use an encoded string as a path?
(https://developers.google.com/maps/documentation/utilities/polylinealgorithm)

**Comment by DebkanchanSamadder** [comment]
> Is it possible to use an encoded string as a path?
> (https://developers.google.com/maps/documentation/utilities/polylinealgorithm)

Yes you just need to have a decoder function. I have one because I use the directions api to do exactly this. let me put it up on a gist.

**Comment by SDRuiz** [comment]
Can I ask some questions regarding the google_maps_flutter polylines? because in the flutter-dev groups noone answer.
Thanks!

**Pull request merged**
